agent/api/container.go

```diff
@@ -32,15 +33,35 @@ const (
 	// that specifies that the log driver should be authenticated using the
 	// execution role
 	awslogsAuthExecutionRole = "ExecutionRole"
+
+	// dockerHealthCheckType is the type of container health check provided by docker
+	dockerHealthCheckType = "docker"
 )
 
 // DockerConfig represents additional metadata about a container to run. It's
 // remodeled from the `ecsacs` api model file. Eventually it should not exist
 // once this remodeling is refactored out.
 type DockerConfig struct {
-	Config     *string `json:"config"`
+	// Config is the configuration used to create container
+	Config *string `json:"config"`
+	// HostConfig is the configuration of container related to host resource
 	HostConfig *string `json:"hostConfig"`
-	Version    *string `json:"version"`
+	// Version specifies the docker client API version to use
+	Version *string `json:"version"`
+	// HealthCheck is the configuration of docker health check
```

Comment:
Is this another json-serialized object?  Is this part of Config/HostConfig?

Reply:
No and it's not part of the Config/HostConfig from the payload message. It was sent as a string and agent will unmarshal it into the docker recognized struct [api/task.go](https://github.com/aws/amazon-ecs-agent/pull/1141/files/dfdc730c25029805e1b602ecb7b64a35f70a6a74#diff-e008975819a900a8494cf91881a8304aR458)

Reply:
Why isn't it part of the `Config` from the payload message?  It's being stuck in the `Config` anyway, so why do we need a separate unmarshal for it?